flake.nix

```diff
@@ -94,157 +92,166 @@
+                    default = false;
+                    description = "Whether to enable slasher.";
+                  };
+                  historyLength = mkOption {
+                    type = types.int;
+                    default = 4096;
+                    description = "Number of epochs to store.";
+                  };
+                  maxDatabaseSize = mkOption {
+                    type = types.int;
+                    default = 256;
+                    description = "Maximum size of the database in gigabytes.";
+                  };
+                };
+                mev-boost = {
+                  enable = mkOption {
+                    type = types.bool;
+                    default = false;
+                    description = "Whether to enable MEV-Boost.";
+                  };
+                  endpoint = mkOption {
+                    type = types.str;
+                    default = "http://127.0.0.1:18550";
+                    description = "Listening interface for MEV-Boost server.";
+                  };
+                };
+                dataDir = mkOption {
+                  type = types.path;
+                  default = "/var/mnt/lighthouse";
+                  description = "Data directory for the databases.";
                 };
-              };
-              datadir = mkOption {
-                type = types.str;
-                default = "/var/mnt/lighthouse";
-                description = "Data directory for the databases";
               };
             };
-          };
 
-          homestakeros = {
+
+          homestaker = {
             system = "x86_64-linux";
             specialArgs = { inherit inputs outputs; };
             modules = [
               ./system
               ./system/ramdisk.nix
               ./system/formats/netboot-kexec.nix
-              self.nixosModules.homestakeros
-              disko.nixosModules.disko
+              self.nixosModules.homestaker
               {
                 system.stateVersion = "23.05";
               }
             ];
           };
         in
         rec {
-          # filters options recursively
-          # option exports -- accessible through 'nix eval --json .#exports'
+          # Filters options to exports recursively
+          # Accessible through 'nix eval --json .#exports'
           exports = nixpkgs.lib.attrsets.mapAttrsRecursiveCond
             (v: ! nixpkgs.lib.options.isOption v)
             (k: v: v.type.name)
-            homestakeros_options;
+            homestakerOptions;
 
           overlays = import ./overlays { inherit inputs; };
 
+          # NixOS configuration entrypoints for the frontend
           nixosConfigurations = with nixpkgs.lib; {
-            "homestakeros" = nixosSystem homestakeros;
+            "homestaker" = nixosSystem homestaker;
           } // (with nixpkgs-stable.lib; { });
 
-          # usage: https://github.com/ponkila/homestaking-infra/commit/574382212cf817dbb75657e9fef9cdb223e9823b
```

Comment:
Can we keep a reference to some pinned version of homestaking-infra which shows how this is used?

Reply:
I think it would be a good idea to provide the reference in the `README.md`. I can add a link under the "Usage as module" section for better access.

Reply:
Ok, README sounds good enough.